workcal/angular/src/app/calendar-page/calendar-page.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, OnInit } from '@angular/core';
+import { EventApiService } from '../event-api.service';
+import { EventDto } from '../models/event-dto.model';
+
+@Component({
+  selector: 'app-calendar-page',
+  templateUrl: './calendar-page.component.html',
+  styleUrls: ['./calendar-page.component.scss']
+})
+export class CalendarPageComponent implements OnInit {
+  events: EventDto[] = [];
+  schedulerEvents: any[] = []; // For DevExtreme Scheduler
+
+  currentDate: Date = new Date();
+  currentView: string = 'day';
+
+  constructor(private eventApiService: EventApiService) { }
+
+  ngOnInit(): void {
+    this.fetchEvents();
+  }
+
+
+
+
+  fetchEvents(): void {
+    this.eventApiService.getAllEvents().subscribe(data => {
+      this.events = data;
+      this.schedulerEvents = data.map(event => ({
+        id:event.id,
+        startDate: event.startTime,
+        endDate: event.endTime,
+        text: event.name,
+        location: event.location
+      }));
+    });
+  }
+
+
+  createEvent(newEvent: EventDto): void {
+    this.eventApiService.createEvent(newEvent).subscribe(
+      () => {
+        console.log('Event created successfully.');
+        this.fetchEvents(); // Refresh events after creating
+      },
+      (error) => {
+        console.error('Error creating event:', error);
+      }
+    );
+  }
+  updateEvent(eventToUpdate: EventDto): void {
+    this.eventApiService.updateEvent(eventToUpdate.id, eventToUpdate).subscribe( () => {
+      console.log('Event updated successfully.');
+      this.fetchEvents(); // Refresh events after updating
+    },
+    (error) => {
+      console.error('Error updating event:', error);
+    }
+  );
+  }
+  deleteEvent(eventId: string): void {
+    this.eventApiService.deleteEvent(eventId).subscribe(() => {
+      const index = this.events.findIndex(event => event.id === eventId);
+      if (index !== -1) {
+        this.events.splice(index, 1);
```

Comment:
thats not how handle delete, you find the element before delete request to backend you start the request and if successfull then splice the element from array and refetch!